lib/cluster.js

```diff
@@ -260,6 +260,9 @@ function masterInit() {
       return process.nextTick(setupSettingsNT, settings);
     initialized = true;
     schedulingPolicy = cluster.schedulingPolicy;  // Freeze policy.
+    if (settings.scheduler) {
+      settings.scheduler(handles);
```

Comment:
Maybe I'm missing something, but this doesn't seem like the right approach to me.